interop-scoring/main.js

```diff
@@ -0,0 +1,506 @@
+  '/webcodecs/videoDecoder-codec-specific.https.any.worker.html?vp9',
+  '/webcodecs/videoDecoder-codec-specific.https.any.worker.html?av1',
+  '/webcodecs/videoFrame-construction.any.html',
+  '/webcodecs/videoFrame-construction.crossOriginSource.sub.html',
+  '/webcodecs/videoFrame-construction.window.html',
+  '/webcodecs/videoFrame-serialization.crossAgentCluster.https.html',
+  '/webcodecs/videoFrame-serialization.crossAgentCluster.https.html',
+  '/webcodecs/temporal-svc-encoding.https.any.html?h264',
+  '/webcodecs/temporal-svc-encoding.https.any.html?vp8',
+  '/webcodecs/temporal-svc-encoding.https.any.html?vp9',
+  '/webcodecs/temporal-svc-encoding.https.any.worker.html?h264',
+  '/webcodecs/temporal-svc-encoding.https.any.worker.html?vp8',
+  '/webcodecs/temporal-svc-encoding.https.any.worker.html?vp9',
+  '/webcodecs/videoFrame-serialization.crossAgentCluster.https.html',
+  '/webcodecs/videoFrame-serialization.crossAgentCluster.https.html',
+  '/webcodecs/videoFrame-serialization.crossAgentCluster.https.html',
+  '/webcodecs/full-cycle-test.https.any.html?av1',
+  '/webcodecs/full-cycle-test.https.any.html?h264_annexb',
+  '/webcodecs/full-cycle-test.https.any.html?h264_avc',
+  '/webcodecs/full-cycle-test.https.any.html?vp9_p0',
+  '/webcodecs/full-cycle-test.https.any.html?vp9_p2',
+  '/webcodecs/full-cycle-test.https.any.worker.html?av1',
+  '/webcodecs/full-cycle-test.https.any.worker.html?h264_annexb',
+  '/webcodecs/full-cycle-test.https.any.worker.html?h264_avc',
+  '/webcodecs/full-cycle-test.https.any.worker.html?vp9_p0',
+  '/webcodecs/full-cycle-test.https.any.worker.html?vp9_p2',
+  '/webcodecs/full-cycle-test.https.any.html?vp8',
+  '/webcodecs/full-cycle-test.https.any.worker.html?vp8',
+  // interop-2023-webcomponents
+  '/shadow-dom/focus/focus-shadowhost-display-none.html',
+  '/custom-elements/form-associated/ElementInternals-labels.html',
+  '/custom-elements/form-associated/ElementInternals-setFormValue.html',
+  '/custom-elements/form-associated/ElementInternals-validation.html',
+  '/custom-elements/form-associated/form-disabled-callback.html',
+]);
+
+
+// Calculate interop score (passing in all browsers) for a category
+// after tracking the category's scores for each browser.
+function aggregateInteropTestScores(interopScores, numBrowsers) {
+  if (interopScores.length === 0) return 0;
+  let aggregateScore = 0;
+  for (const [, results] of interopScores) {
+    let subtestsAllPassing = 0;
+    for (const [, subtestNumPassing] of results) {
+      // A subtest counts toward the interop score if every browser passed the subtest.
+      // In other words, the number of passes should match the number of browsers.
+      if (subtestNumPassing === numBrowsers) {
+        subtestsAllPassing += 1;
+      }
+    }
+    aggregateScore += Math.floor(1000 * subtestsAllPassing / results.size);
+  }
+  return Math.floor(aggregateScore / interopScores.size) || 0;
+}
```

Comment:
This is a major new section of the script, which calculates the interop scores for each category.